Network bindings for GDNative

Faless opened the pull request:
Support `PacketPeer`, `NetworkedMultiplayerPeer`, and `StreamPeer` (which I strongly suggest be renamed just `MultiplayerPeer`, we can keep the alias in the class DB for compatibility).

~I still want to add:~
- ~`StreamPeer`~ Done
- ~`MultiplayerAPI` (to implement custom protocol, custom serialization, etc).~ Will go in a separate PR, likely only need generic scripting binding, needs some investigation.

The whole process is a bit boring as I feel that it might be somehow automated. In the meantime, I'd like an opinion from @karroffel . EDIT: **Now stable and ready for review to anyone interested**

You can test it with:
https://github.com/Faless/GDNativeNet , btw, great job on the new C++ bindings! They are a lot more godot-y :+1:

karroffel reviewed (approved):
Looking good (but needs a rebase :stuck_out_tongue: )

Faless committed:
Add [Packet/Stream/Multiplayer]PeerGDNative

They provide an interface to implement PacketPeer, StreamPeer, and
NetworkedMultiplayerPeer via GDNative.

Faless commented:
Rebased on an older master commit (before new AnimationEditor that is still work in progress).
Added StreamPeerGDNative.
I think this is good to go, MultiplayerAPI bindings will go in a separate PR as I still have to decide how to expose it's functions best.

karroffel commented on modules/gdnative/include/net/godot_net.h:
```diff
@@ -0,0 +1,118 @@
+
+// Use these to populate version in your plugin
+#define GODOT_NET_API_MAJOR 3
+#define GODOT_NET_API_MINOR 1
```
All the other plugins start from 1.0, since that's extension version number, not Godot version number. But it doesn't really make a huge difference, you are free to use any other versioning scheme (as long as it's semantic versioning)

akien-mga replied:
@Faless If you want to change that to use 1.0, it's now or never ;)

Faless replied:
Whoops, late to the party, I had a very busy day, But I do like the idea of having the GDNative Network Version in sync with the Godot one, so all good in the end :)

karroffel reviewed (approved):
Still looks fine :)
